File: python/examples/eg1.py
from searchsorted._internal import cartesian_3d, searchsorted_3d 


sample = [('a1','b2','c1'),('a2','b3','c2')]

dim1 = ['a1','a2']
dim2 = ['b1','b2','b3']
dim3 = ['c1','c2']

full = cartesian_3d(dim1,dim2,dim3)

index = searchsorted_3d(full, sample)

print(f"Location of sample tuples in full: {index}")

File: examples/examples/eg1.rs
use searchsorted::{cartesian_3d,searchsorted_3d};

fn main() {
    let sample = vec![(String::from("a1"),String::from("b2"),String::from("c1")),(String::from("a2"),String::from("b3"),String::from("c2"))];

    let dim1 = vec![String::from("a1"),String::from("a2")];
    let dim2 = vec![String::from("b1"),String::from("b2"),String::from("b3")];
    let dim3 = vec![String::from("c1"),String::from("c2")];

    let full = cartesian_3d(dim1,dim2,dim3);

    let index = searchsorted_3d(full, sample);

    println!("Location of sample tuples in full: {:?}", index)
}